### PR TITLE
Refactor tiny version parsing

### PR DESCRIPTION
**Pull request by thaJeztah**
Had this branch still lying around on my machine, haha, and noticed an issue in the latest version-parsing change.

**- What I did**

Invalid version strings for the init (tini) binary were still accepted, which lead to (e.g.)

    "hello world"

Being used as `"vhello world"`

This makes the version parsing slightly stricter

Also refactored the "init" version parsing; extracted it to a function and added a unit-test

relates to https://github.com/docker/docker/pull/32187

ping @tianon @vdemeester PTAL

### CODE REVIEW

File: daemon/info_unix.go [hunk 1]
```diff
@@ -54,32 +55,37 @@ func (daemon *Daemon) FillPlatformInfo(v *types.Info, sysInfo *sysinfo.SysInfo)
...
+	parts := strings.Split(strings.TrimSpace(v), " - ")
+
+	if len(parts) >= 2 {
+		gitParts := strings.Split(parts[1], ".")
+		if len(gitParts) == 2 && gitParts[0] == "git" {
+			version.ID = gitParts[1]
+			version.Expected = dockerversion.InitCommitID[0:len(version.ID)]
+		}
+	}
+	if version.ID == "" && strings.HasPrefix(parts[0], "tini version") {
```
Comment (tianon) [inline daemon/info_unix.go]: Should this include the trailing space too, as in the `TrimPrefix` below? (`strings.HasPrefix(parts[0], "tini version ")`)

**Comment by tianon** [comment]
Just one minor question -- otherwise LGTM :+1: :heart:

File: daemon/info_unix.go (hunk 1 above)
Comment (thaJeztah) [inline daemon/info_unix.go]: ah yeah was thinking that; didn't think it'd make a big difference but can update if you want. Nowhere near my computer now though haha

Comment (tianon) [inline daemon/info_unix.go]: It's probably safer to do so -- that way if all we get is `tini version` (by itself) for some reason, it won't give us `vtini version` later and we'll get `N/A` instead like we expect (a new test case :laughing:). :+1:

**Review by vdemeester — APPROVED** [review]
LGTM 😻

File: daemon/info_unix.go (hunk 1 above)
Comment (thaJeztah) [inline daemon/info_unix.go]: ah didn't think of that corner. I'll update when i'm back at my computer

**Comment by mlaventure** [comment]
LGTM once tianon change has been added :)

**Comment by thaJeztah** [comment]
Looks like `_unix_test.go` also runs on Windows; will have to fix that

```
23:57:24 # github.com/docker/docker/daemon
23:57:24 daemon\info_unix_test.go:19: undefined: dockerversion.InitCommitID
23:57:25 ok  	github.com/docker/docker/cmd/dockerd	0.148s	coverage: 12.5% of statements
```

**Comment by thaJeztah** [comment]
t.b.h., actually considered passing "expected" version to be passed as an argument, but i'll have a look

File: daemon/info_unix_test.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// +build !windows
```
Comment (thaJeztah) [inline daemon/info_unix_test.go]: added this to fix CI :smile:

**Comment by thaJeztah** [comment]
all green, gonna do a self-five, and merge :smile:

**Pull request merged**